iree/compiler/Conversion/LinalgToSPIRV/VectorToGPUPass.cpp

```diff
@@ -293,6 +293,44 @@ class VectorContractLowering : public OpRewritePattern<vector::ContractionOp> {
   }
 };
 
+// Lower elementwise operation from N-D vector to 1-D vectors that can be
+// natively supported.
+class ElementwiseLowering : public RewritePattern {
```

Comment:
Is there a reason this is not upstreamed? In other words this seems to be a "vector unroll" transformation too right?

Reply:
I was thinking about it but I didn't know where to move it. It is not really unrolling, it is just merging all the vector dimension into 1 (although for IREE we expect that all the higher dimensions are 1). I had talked to other projects that needed similar things but I'm not really sure where it belongs upstream or what would be a good generalization (here is it limited to elementwise but it could apply to more things)

Reply:
Ok, got thrown off by the comment of lowering N-D vectors to 1D vector. I would like to understand for my own skae why all the higher-dimensions are 1. Fine to land it here, but is as good as landing upstream  and giving it whatever name you want.

Reply:
The higher dimensions are all 1 because we picked 1 as unrolling shape. When we unroll we basically pick a dimension like <1x1x4xf32> (rank depends on original shape). So we end up with ops like:
add ... : <1x1x4xf32> 
This converts it to:
add ... : <4xf32>